File: 2268-minimum-number-of-keypresses/2268-minimum-number-of-keypresses.rb
# @param {String} s
# @return {Integer}
def minimum_keypresses(s)
    n = s.length
    if n <=9
        return n
    end
    
    x = Array.new(10, 0)
    count = Array.new(26, 0)
    
    (0..n-1).each do |i|
        idx = s[i].ord-'a'.ord
        count[idx]+=1
    end
    
    temp = []
    (0..25).each do |i|
        if count[i]!=0
            temp << count[i]
        end
    end
    
    temp.sort! { |a, b| b<=>a }

    pos=1
    sz = temp.size
    ans=0
    
    (0..sz-1).each do |i|
        x[pos%9] += 1
        ans += (x[pos%9]*temp[i])
        pos+=1
    end
    
    return ans
    
end

File: 2268-minimum-number-of-keypresses/2268-minimum-number-of-keypresses.cpp
class Solution {
public:
    int minimumKeypresses(string s) {
        if(s.length() <= 9) return s.length();
        int ans=0;
        vector<int> m(10, 0);
        vector<int> y(26, -1);
        vector<int> l(26, 0);
        
        vector<int> temp;
        
        for(auto x:s) l[x-'a']+=1;
        for(auto y:l) {
            if(y!=0) temp.push_back(y);
        }
        
        sort(temp.begin(), temp.end(), [](const int& a, const int& b){
            return a > b;
        });
        
        int pos=1;
        
        for(int i=0;i<temp.size();i++){
            int count = temp[i];
            
            m[pos%9] +=1;
            ans += (m[pos%9]*count);
            pos+=1;
            
        }
        
        
        
        return ans;
        
    }
};